### PR TITLE
Dev. Remade import to run code

### PR DESCRIPTION
**Pull request by VadimH1**
After your help with change import I remade import to run code. first_app.models, first_app.hello -ModuleError, app is not defined.

### CODE REVIEW

File: first_app/db.py [hunk 1]
```diff
@@ -15,46 +16,41 @@
...
-    click.echo('Initialized the database.')
-
-def init_app(app):
-    app.teardown_appcontext(close_db)
-    app.cli.add_command(init_db_command)
+db = SQLAlchemy()
+
+# db = SQLAlchemy(app)
+# migrate = Migrate(app, db)
+
```
Comment (AndyKovv) [inline first_app/db.py]: Remove comments

File: first_app/hello.py [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
 from functools import wraps
-from flask import g, request, redirect, url_for
-from .models import Post, User
-from .db import get_db, db_session
+from flask import g, redirect, url_for
+from .models import Post, User, Comments
 from . import db
+from werkzeug.utils import secure_filename
+
+# app = create_app()
```
Comment (AndyKovv) [inline first_app/hello.py]: remove

File: first_app/hello.py [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
 
 hello_urls = Blueprint("sync", __name__)
```
Comment (AndyKovv) [inline first_app/hello.py:16]: rename to name

File: first_app/hello.py [hunk 4]
```diff
@@ -91,13 +93,11 @@ def login_api():
 
     # user_id має знаходитись в середині JWT.
     token_data = {"user_id": user["id"]}
-    access_token = jwt.encode(token_data, app.config['SECRET_KEY'], algorithm='HS256')
+    access_token = jwt.encode(token_data, app.config['SECRET_KEY'], algorithm='HS256')
```
Comment (AndyKovv) [inline first_app/hello.py]: app.config['SECRET_KEY'] get this from separate file.

File: first_app/hello.py [hunk 5]
```diff
@@ -109,42 +109,49 @@ def user_info_api(user_id):
...
+    title = data['title']
+    body = data['body']
+    created = data['create']	
+    author_id = None
+
+    post = Post(title, body, created, author_id)
+    db.session.add(post)
+    db.session.commit()
+
+    return "Create", 200
```
Comment (AndyKovv) [inline first_app/hello.py]: {"status": "OK"}

File: first_app/hello.py [hunk 6]
```diff
@@ -109,42 +109,49 @@ def user_info_api(user_id):
...
+    post = Post.query.get(user_id)
     db.session.delete(post)
     db.session.commit()
         
-    return "Post deleted", 200
-    
+    return "Deleted", 200
+
+
+@hello_urls.route('/api/v1/update-post/<user_id>', methods=['PUT'])
```
Comment (AndyKovv) [inline first_app/hello.py]: /<post_id>/<user_id>

File: first_app/hello.py [hunk 7]
```diff
@@ -109,42 +109,49 @@ def user_info_api(user_id):
...
+@login_required
+def update_post(user_id):
+    """Редагування постів"""
+    data = request.json
+    title = data['title']
```
Comment (AndyKovv) [inline first_app/hello.py:133]: post = Post.query.filter(Post.id == post_idm, Post.author_id == user.id).first()
if not post:
    return {"error": "Post not found"}, 400
Post.query.update()

File: first_app/hello.py [hunk 8]
```diff
@@ -109,42 +109,49 @@ def user_info_api(user_id):
...
+    body = data['body']
+    created = data['create']	
+    author_id = None
+
+    post = Post.query.get(user_id)
```
Comment (AndyKovv) [inline first_app/hello.py]: Post.query.update(**data)

File: first_app/hello.py [hunk 9]
```diff
@@ -153,12 +160,108 @@ def api_for_who_i_am(user_id):
...
 
  
-@hello_urls.route('/api/v1/<user_id>/posts/<post_id>', methods=['GET'])
-def get_user_posts(post_id):
-    """Виведення постів по користувачу. Виводимо пост користувача"""
+@hello_urls.route('/api/v1/<user_id>/posts/', methods=['GET'])
+@login_required
+def get_user_post(user_id):
+    """Список постів юзера"""
+    post = Post.query.all(user_id)
```
Comment (AndyKovv) [inline first_app/hello.py]: posts = Post.query.filter(Post.author_id == user_id).all()

File: first_app/hello.py [hunk 10]
```diff
@@ -153,12 +160,108 @@ def api_for_who_i_am(user_id):
...
+    return "Отримали пости юзера", 200
```
Comment (AndyKovv) [inline first_app/hello.py]: user_posts = []
for post in posts:
  _post_data = {
     "id": post.id,
}
    user_posts.append(_post_data)
return user_posts, 200

File: first_app/hello.py [hunk 11]
```diff
@@ -153,12 +160,108 @@ def api_for_who_i_am(user_id):
...
+
+
+@hello_urls.route('/api/v1/<user_id>/post/<post_id>', methods=['GET'])
+@login_required
+def user_post(post_id):
+    """Виводимо пост користувача"""
     post = Post.query.get(post_id)
```
Comment (AndyKovv) [inline first_app/hello.py]: post  = Post.query.filter(Post.id == post_id, Post.author_id == user_id).one()
return {
"id": post.id,
"title": post.title,
......
}

File: first_app/hello.py [hunk 12]
```diff
@@ -153,12 +160,108 @@ def api_for_who_i_am(user_id):
...
+    _phone_number = data['phone_number']
+    _first_name = data['first_name']
+    _second_name = data['second_name']
+
+    user_info = User.query.filter(User.phone_number==_phone_number).one()
+
+    if user_info is _phone_number:
+        return (f'This {_phone_number} is already exists')
+    
+    new_info = User(phone_number=_phone_number, first_name=_first_name, second_name=_second_name)
```
Comment (AndyKovv) [inline first_app/hello.py:200]: Update.

File: first_app/hello.py [hunk 13]
```diff
@@ -153,12 +160,108 @@ def api_for_who_i_am(user_id):
...
+    db.session.add(new_info)
+    db.commit()
+
+    return {}, 200
```
Comment (AndyKovv) [inline first_app/hello.py]: {updated_user_info}

File: first_app/hello.py [hunk 14]
```diff
@@ -153,12 +160,108 @@ def api_for_who_i_am(user_id):
...
+
+
+@hello_urls.route('/api/vi/user-info/change-password', methods=['PUT'])
+@login_required
+def change_user_password(user_id):
```
Comment (AndyKovv) [inline first_app/hello.py:210]: user_id should be in a url

File: first_app/hello.py [hunk 15]
```diff
@@ -153,12 +160,108 @@ def api_for_who_i_am(user_id):
...
+    old_password = data['password']
+    new_password = data['new_password']
+
+    user = User.query.filter(User.id==user_id)
+
+    if old_password == user.password:
+    # if check_password_hash(password_hash, password)    
+        return {'Password confirmed. You can change new password'}
+    
+    new_password = User(new_password=generate_password_hash(new_password, method='sha256'))
```
Comment (AndyKovv) [inline first_app/hello.py]: Update action

File: first_app/hello.py [hunk 16]
```diff
@@ -153,12 +160,108 @@ def api_for_who_i_am(user_id):
...
+    db.session.add(new_password)
+    db.commit()
+
+    return "Password changed", 200
```
Comment (AndyKovv) [inline first_app/hello.py]: {}

File: first_app/hello.py [hunk 17]
```diff
@@ -153,12 +160,108 @@ def api_for_who_i_am(user_id):
...
+
+
+@hello_urls.route('/api/v1/create-comments/<user_id>', methods=['POST'])
```
Comment (AndyKovv) [inline first_app/hello.py:224]: user_id, post_id

File: first_app/hello.py [hunk 18]
```diff
@@ -153,12 +160,108 @@ def api_for_who_i_am(user_id):
...
+def create_comment(user_id):
+    """Додавання нового коментаря"""
+    data = request.json
+    text = data['text']
+    created = data['created']
+    new_comm = Comments(text, created)
```
Comment (AndyKovv) [inline first_app/hello.py]: post_id  == ForeignKey("post.id")

File: first_app/hello.py [hunk 19]
```diff
@@ -153,12 +160,108 @@ def api_for_who_i_am(user_id):
...
+    
+    db.session.add(new_comm)
+    db.commit()
+
+    comment = Comments.query.get(new_comm.id)
+    return "Comment created", 200
```
Comment (AndyKovv) [inline first_app/hello.py]: {**comment_data}

File: first_app/hello.py [hunk 20]
```diff
@@ -153,12 +160,108 @@ def api_for_who_i_am(user_id):
...
+
+
+@hello_urls.route('/api/v1/update-comments/<user_id>', methods=['PUT'])
```
Comment (AndyKovv) [inline first_app/hello.py]: comment_id

File: first_app/hello.py [hunk 21]
```diff
@@ -153,12 +160,108 @@ def api_for_who_i_am(user_id):
...
+
+    comment = Comments.query.get(new_comm.id)
+    return "Comment created", 200
+
+
+@hello_urls.route('/api/v1/update-comments/<user_id>', methods=['PUT'])    
+def update_comment(user_id):
+    """Редагування(оновлення) коментарів"""
+    data = request.json
+    comment = Comments.query.get(user_id)
```
Comment (AndyKovv) [inline first_app/hello.py]: Check  if comment related to user.
Update comment.

File: first_app/hello.py [hunk 22]
```diff
@@ -153,12 +160,108 @@ def api_for_who_i_am(user_id):
...
+    created = data['created']
+
+    comment.text = text
+    comment.created = created
+
+    db.session.commit()
+    return "Comment updated", 200
+
+
+@hello_urls.route('/api/v1/delete-comments', methods=['DELETE'])
```
Comment (AndyKovv) [inline first_app/hello.py]: comment_id

File: first_app/hello.py [hunk 23]
```diff
@@ -153,12 +160,108 @@ def api_for_who_i_am(user_id):
...
+    return "Comment deleted", 200
+
+
+@hello_urls.route('/api/v1/upload-files', methods=['POST'])
+def upload_files():
+    uploaded_files = request.files['file']
+    if uploaded_files.filename != '':
+        uploaded_files.save(secure_filename(uploaded_files.filename))
+
+    return "Added images"
```
Comment (AndyKovv) [inline first_app/hello.py]: return uploaded file_id {"id": file.id}

File: first_app/hello.py [hunk 24]
```diff
@@ -153,12 +160,108 @@ def api_for_who_i_am(user_id):
...
+    db.commit()
+
+    return "Comment deleted", 200
+
+
+@hello_urls.route('/api/v1/upload-files', methods=['POST'])
+def upload_files():
+    uploaded_files = request.files['file']
+    if uploaded_files.filename != '':
+        uploaded_files.save(secure_filename(uploaded_files.filename))
```
Comment (AndyKovv) [inline first_app/hello.py:276]: path_to_file = uploaded_files.save(secure_filename(uploaded_files.filename))
class FileUpload(db.Model):
   id = Int
   url = Str

file = FileUpload(url = path_to_file)
db.session.add(file)
db.session.commit()
{"id": file.id}

**Review by AndyKovv — CHANGES_REQUESTED** [review]

File: first_app/db.py (hunk 1 above)
Comment (VadimH1) [reply first_app/db.py]: Removed

File: first_app/hello.py (hunk 2 above)
Comment (VadimH1) [reply first_app/hello.py]: Removed

File: first_app/hello.py (hunk 5 above)
Comment (VadimH1) [reply first_app/hello.py]: changed

File: first_app/hello.py (hunk 20 above)
Comment (VadimH1) [reply first_app/hello.py]: changed

File: first_app/hello.py (hunk 21 above)
Comment (VadimH1) [reply first_app/hello.py]: added check

File: first_app/hello.py (hunk 22 above)
Comment (VadimH1) [reply first_app/hello.py]: changed

File: first_app/hello.py (hunk 7 above)
Comment (VadimH1) [reply first_app/hello.py:133]: changed

File: first_app/hello.py (hunk 8 above)
Comment (VadimH1) [reply first_app/hello.py]: changed

File: first_app/hello.py (hunk 18 above)
Comment (VadimH1) [reply first_app/hello.py]: Why I must to write ForeignKey?

File: first_app/hello.py (hunk 19 above)
Comment (VadimH1) [reply first_app/hello.py]: changed

File: first_app/hello.py (hunk 17 above)
Comment (VadimH1) [reply first_app/hello.py:224]: changed

File: first_app/hello.py [hunk 25]
```diff
@@ -37,7 +38,7 @@ def register_user_api():
     second_name = data["second_name"]
     password = data["password"]
     
-    user = User('phone_number', 'first_name', 'second_name', 'password')
+    user = User("phone_number", "first_name", "second_name", "password")
```
Comment (AndyKovv) [inline first_app/hello.py]: Should be User(phone_number, first_name, second_name, password). Why you use strings here?

File: first_app/hello.py [hunk 26]
```diff
@@ -109,56 +104,193 @@ def user_info_api(user_id):
...
     
     
-@hello_urls.route('/api/v1/delete-post/<user_id>', methods=['GET','POST'])
-def delete_posts(post_id):
-    post = Post.query.filter_by(id=post_id, user_id=User.id).first()
+@hello_urls.route('/api/v1/delete-post/<user_id>', methods=['DELETE'])
+@login_required
+def delete_post(user_id):
+    """Видалення постів"""
+    post = Post.query.get(user_id)
```
Comment (AndyKovv) [inline first_app/hello.py:121]: This is works? I think should be Post.query.get(Post.author_id == user_id)

File: first_app/hello.py [hunk 27]
```diff
@@ -109,56 +104,193 @@ def user_info_api(user_id):
...
+
+@hello_urls.route('/api/v1/update-post/<post_id>/<user_id>', methods=['PUT'])
+@login_required
+def update_post(user_id, post_id):
+    """Оновлення постів"""
+    data = request.json
+    title = data['title']
+    body = data['body']
+    created = data['create']	
+    author_id = None
```
Comment (AndyKovv) [inline first_app/hello.py]: author_id = g.request.user_id

File: first_app/hello.py [hunk 28]
```diff
@@ -109,56 +104,193 @@ def user_info_api(user_id):
...
+
+    post = Post.query.filter(Post.id == post_id, Post.author_id == user_id).first()
```
Comment (AndyKovv) [inline first_app/hello.py:138]: post = Post.query.filter(Post.id == post_id, Post.author_id == author_id).first()

File: first_app/hello.py [hunk 29]
```diff
@@ -109,56 +104,193 @@ def user_info_api(user_id):
...
+    post_data = {
+        "id": post.id,
+        "author_id": post.author_id,
+        "title": post.title,
+        "body": post.body,
+        "created": post.created
+    }
+    user_post.append(post_data)
+
+    return {user_post}, 200
```
Comment (AndyKovv) [inline first_app/hello.py]: ? this will raise an error. Have you test this API already?

File: first_app/hello.py [hunk 30]
```diff
@@ -109,56 +104,193 @@ def user_info_api(user_id):
...
+
+
+@hello_urls.route('/api/v1/<user_id>/post/<post_id>', methods=['GET'])
+@login_required
+def user_post(post_id, user_id):
```
Comment (AndyKovv) [inline first_app/hello.py]: def user_post(user_id, post_id):

File: first_app/hello.py [hunk 31]
```diff
@@ -109,56 +104,193 @@ def user_info_api(user_id):
...
+@login_required
+def edit_user_info(user_id):
+    data = request.json
+    _phone_number = data['phone_number']
+    _first_name = data['first_name']
+    _second_name = data['second_name']
+
+    user_info = User.query.filter(User.phone_number==_phone_number).one()
+
+    if user_info is _phone_number:
```
Comment (AndyKovv) [inline first_app/hello.py]: this is not Correct, You can't check user_info and _phone_number. Check this api through postman.

File: first_app/hello.py (hunk 25 above)
Comment (VadimH1) [reply first_app/hello.py]: Yes, I changed mistake already

File: first_app/hello.py (hunk 26 above)
Comment (VadimH1) [reply first_app/hello.py:121]: Yes, this is working. I checked code with Danila yesterday. Post deleted from DB.

File: first_app/hello.py (hunk 28 above)
Comment (VadimH1) [reply first_app/hello.py:138]: But code was working with author_id == user_id

File: first_app/hello.py (hunk 29 above)
Comment (VadimH1) [reply first_app/hello.py]: I changed problems and rewrote the code:
def get_user_post(user_id):
    """Список постів юзера"""
    user_post = []
    post = Post.query.filter(Post.author_id==user_id).all()
    for i in post:
        user_post.append({
            "author_id": i.author_id,
            "title": i.title,
            "body": i.body,
            "created": i.created
        })

    return {"user_list_post": user_post}, 200

File: first_app/hello.py (hunk 30 above)
Comment (VadimH1) [reply first_app/hello.py]: +

File: first_app/hello.py (hunk 31 above)
Comment (VadimH1) [reply first_app/hello.py]: This check code is not working and I just commented it

**Pull request merged**